narrative_llm_agent/agents/analyst.py

```diff
@@ -1,27 +1,67 @@
 from .kbase_agent import KBaseAgent
 from crewai import Agent
 from langchain_core.language_models.llms import LLM
+from langchain_openai import OpenAIEmbeddings
+from langchain.pydantic_v1 import BaseModel, Field
+from langchain.vectorstores import Chroma
+from langchain.memory import ConversationBufferMemory, ReadOnlySharedMemory
+from langchain.chains import RetrievalQA
+from langchain.tools import BaseTool, tool
+
+
+class AnalystInput(BaseModel):
+    input: str = Field(description="query to look up KBase documentation, catalog or tutorials")
 
 class AnalystAgent(KBaseAgent):
     role="Computational Biologist and Geneticist"
     goal="Analyze and interpret datasets, and make suggestions into next analysis steps."
     backstory="""You are an expert academic computational biologist with decades of
     experience working in microbial genetics. You have published several genome announcement
-    papers and have worked extensively with novel sequence data. You understand the most
-    common workflows for assembling new genomes from sequence data. You have implemented and
-    published these workflows in the KBase Narrative Interface for use by yourself and others.
-    You are highly motivated to produce deep analytical insights from data results."""
+    papers and have worked extensively with novel sequence data."""
 
     def __init__(self: "AnalystAgent", token: str, llm: LLM):
         super().__init__(token, llm)
         self.__init_agent()
 
-    def __init_agent(self):
+    def __init_agent(self: "AnalystAgent"):
+
+        @tool("Kbase documentation retrieval tool", args_schema = AnalystInput, return_direct=True)
```

Comment:
KBase is usually spelled with K B capitalized. I'm not sure if that'll confuse the LLMs or not.